vip_hci/hci_dataset.py

```diff
@@ -789,16 +793,101 @@ def inject_companions(self, flux, rad_dists, n_branches=1, theta=0,
             if self.wavelengths is None:
                 raise ValueError('The wavelengths vector has not been set')
 
-        self.cube, yx = cube_inject_companions(self.cube, self.psfn,
-                                               self.angles, flux, self.px_scale,
-                                               rad_dists, n_branches, theta,
-                                               imlib, interpolation,
-                                               full_output=True,
-                                               verbose=verbose)
+        self.cube, yx = cube_inject_companions(
+            self.cube, self.psfn, self.angles, flux, self.px_scale,
+            rad_dists, n_branches, theta, imlib, interpolation,
+            full_output=True, verbose=verbose
+        )
+
+        if not hasattr(self, "injections_yx"):
+            self.injections_yx = []
```

Comment:
Let's try to avoid creating attributes outside __init__. It's more clear when we have all the attributes in __init__, even if they are initialized with None or something else

Reply:
👍